diff --git a/Telegram/SourceFiles/data/data_stories.cpp b/Telegram/SourceFiles/data/data_stories.cpp
--- a/Telegram/SourceFiles/data/data_stories.cpp
+++ b/Telegram/SourceFiles/data/data_stories.cpp
@@ -7,6 +7,8 @@
 */
 #include "data/data_stories.h"
 
+#include "api/api_report.h"
+#include "base/options.h"
 #include "base/unixtime.h"
 #include "apiwrap.h"
 #include "core/application.h"
@@ -48,6 +50,13 @@
 constexpr auto kPollingViewsPerPage = Story::kRecentViewersMax;
 
 using UpdateFlag = StoryUpdate::Flag;
+
+base::options::toggle DisableStories({
+	.id = kOptionDisableStories,
+	.name = "Disable stories",
+	.description = "",
+	.defaultValue = false,
+});
 
 [[nodiscard]] std::optional<StoryMedia> ParseMedia(
 		not_null<PeerData*> peer,
@@ -158,6 +167,8 @@
 	}
 	return result;
 }
+
+const char kOptionDisableStories[] = "disable-stories";
 
 StoriesSourceInfo StoriesSource::info() const {
 	return {
@@ -426,6 +437,10 @@
 void Stories::parseAndApply(
 		const MTPPeerStories &stories,
 		ParseSource source) {
+	if (DisableStories.value()) {
+		return;
+	}
+
 	const auto &data = stories.data();
 	const auto peerId = peerFromMTP(data.vpeer());
 	const auto already = _readTill.find(peerId);
@@ -711,6 +726,10 @@
 }
 
 void Stories::loadMore(StorySourcesList list) {
+	if (DisableStories.value()) {
+		return;
+	}
+
 	const auto index = static_cast<int>(list);
 	if (_loadMoreRequestId[index] || _sourcesLoaded[index]) {
 		return;
diff --git a/Telegram/SourceFiles/data/data_stories.h b/Telegram/SourceFiles/data/data_stories.h
--- a/Telegram/SourceFiles/data/data_stories.h
+++ b/Telegram/SourceFiles/data/data_stories.h
@@ -37,6 +37,8 @@
 class Story;
 struct StoryAlbum;
 class StoryPreload;
+
+extern const char kOptionDisableStories[];
 
 struct StoriesIds {
 	std::vector<StoryId> list; // flat_set<int, greater> for saved/archive.
diff --git a/Telegram/SourceFiles/settings/settings_experimental.cpp b/Telegram/SourceFiles/settings/settings_experimental.cpp
--- a/Telegram/SourceFiles/settings/settings_experimental.cpp
+++ b/Telegram/SourceFiles/settings/settings_experimental.cpp
@@ -26,6 +26,7 @@
 #include "core/launcher.h"
 #include "core/sandbox.h"
 #include "chat_helpers/tabbed_panel.h"
+#include "data/data_stories.h"
 #include "dialogs/dialogs_widget.h"
 #include "dialogs/ui/dialogs_layout.h"
 #include "history/history_item_components.h"
@@ -204,6 +205,7 @@
 			rpl::duplicate(reloadOptionsRequests));
 	};
 
+	addToggle(Data::kOptionDisableStories);
 	addToggle(ChatHelpers::kOptionTabbedPanelShowOnClick);
 	addToggle(Dialogs::kOptionForumHideChatsList);
 	addToggle(Dialogs::Ui::kOptionDialogsMuteIcon);
